dev/NavigationView/NavigationView.cpp

```diff
@@ -1513,35 +1511,48 @@ void NavigationView::UpdatePaneLayout()
+                                }
+                                return footerItemsRepeater.ActualHeight() + footerItemsRepeaterTopBottomMargin;
                             }();
 
                             const auto paneFooterActualHeight = [this]() {
                                 if (const auto& paneFooter = m_leftNavFooterContentBorder.get())
                                 {
-                                    const auto paneFooterMargin = paneFooter.Margin();
-                                    return paneFooter.ActualHeight() + paneFooterMargin.Top + paneFooterMargin.Bottom;
+                                    double paneFooterTopBottomMargin = 0.0;
+                                    if (paneFooter.Visibility() == winrt::Visibility::Visible)
+                                    {
+                                        const auto paneFooterMargin = paneFooter.Margin();
+                                        paneFooterTopBottomMargin = paneFooterMargin.Top + paneFooterMargin.Bottom;
+                                    }
+                                    return paneFooter.ActualHeight() + paneFooterTopBottomMargin;
                                 }
                                 return 0.0;
                             }();
 
-                            const auto visualItemsSeparatorHeight = [this]() {
-                                if (const auto& visualItemsSeparator = m_visualItemsSeparator.get())
+                            const auto menuItemsTopBottomMargin = [this, menuItems]() {
+                                if (menuItems.Visibility() == winrt::Visibility::Visible)
                                 {
-                                    const auto visualItemsSeparatorMargin = visualItemsSeparator.Margin();
-                                    return visualItemsSeparator.ActualHeight() + visualItemsSeparatorMargin.Top + visualItemsSeparatorMargin.Bottom;
+                                    const auto menuItemsMargin = menuItems.Margin();
+                                    return menuItemsMargin.Top + menuItemsMargin.Bottom;
                                 }
                                 return 0.0;
                             }();
 
-                            // Footer, PaneFooter, and VisualItemsSeparator are included in the footerGroup to calculate available height for menu items.
-                            const auto footerGroupActualHeight = footersActualHeight + paneFooterActualHeight + visualItemsSeparatorHeight;
+                            // This is the value computed during the measure pass of the layout process. This will be the value used to determine
+                            // the partition logic between menuItems and footerGroup, since the ActualHeight may be taller if there's more space.
+                            const auto menuItemsDesiredHeight = menuItems.DesiredSize().Height + menuItemsTopBottomMargin;
```

Comment:
Unlike ActualHeight/ActualWidth, UIElement.DesiredSize already includes the Margin. So it should be:

`const auto menuItemsDesiredHeight = menuItems.DesiredSize().Height;`